Source/ThirdParty/TurboBadger/tb_atomic_widgets.cpp

```diff
@@ -283,4 +285,355 @@ void TBBarGraph::SetAxis(AXIS axis)
 
 TB_WIDGET_FACTORY(TBBarGraph, TBValue::TYPE_FLOAT, WIDGET_Z_TOP) {}
 
+
+// == TBPromptWindow =======================================
+
+
+TBPromptWindow::TBPromptWindow(TBWidget *target, TBID id)
+    : m_target(target)
+{
+    TBWidgetListener::AddGlobalListener(this);
+    SetID(id);
+}
+
+TBPromptWindow::~TBPromptWindow()
+{
+    TBWidgetListener::RemoveGlobalListener(this);
+    if (TBWidget *dimmer = m_dimmer.Get())
+    {
+        dimmer->GetParent()->RemoveChild(dimmer);
+        delete dimmer;
+    }
+}
+
+bool TBPromptWindow::Show(const char *title, const char *message,
+    const char *preset, int dimmer,
+    int width, int height)
+{
+    TBWidget *target = m_target.Get();
+    if (!target)
+        return false;
+
+    TBWidget *root = target->GetParentRoot();
+
+    const char *source = "TBLayout: axis: y, distribution: gravity, position: left\n"
+    "	TBLayout: axis: y, distribution: gravity, distribution-position: left\n"
+    "		TBTextField: id: 1, gravity: left right\n"
+    "			font: size: 14dp\n"
+    "		TBEditField: id: 2, multiline: 0, styling: 0, adapt-to-content: 0, gravity: left right\n"
+     "			font: size: 14dp\n"
+   "	TBSeparator: gravity: left right\n"
+    "	TBLayout: distribution: gravity\n"
+    "		TBButton: text: \"  OK  \", id: \"TBPromptWindow.ok\"\n"
+    "			font: size: 14dp\n"
+    "		TBButton: text: \"Cancel\", id: \"TBPromptWindow.cancel\"\n"
+    "			font: size: 14dp\n";
```

Comment:
Faced with the options of loading the .ui.txt file or baking it in, baking seemed to be the better choice, less things to go wrong (I had the TBIDs and events to do that).